etna/search/views.py

```diff
@@ -177,192 +213,398 @@ def catalogue_search(request):
-            filter_keyword=filter_keyword,
-            filter_aggregations=get_api_filters_from_form_values(form),
-            stream=Stream.EVIDENTIAL,
-            aggregations=[
-                f"{api_aggregation_name}:{AGGREGATION_SIZE}",
-            ],
-            opening_start_date=opening_start_date,
-            opening_end_date=opening_end_date,
-            sort_by=sort_by,
-            sort_order=SortOrder.ASC,
-            template=Template.DETAILS,
-        )
-        bucket_count_response, result_response = response["responses"]
-        update_field_choices_to_refelect_api_response(
-            form, result_response["aggregations"]
+    base_title = "Search results"
+
+    http_method_names = ["get", "head"]
+
+    def get(self, request: HttpRequest, **kwargs: Any) -> HttpResponse:
+        """
+        Handle GET requests: instantiate a form instance using
+        request.GET as the data, then check if it's valid.
+        """
+        form = self.form = self.get_form()
+        is_valid = form.is_valid()
+        self.api_result = None
+        self.current_bucket_key = form.cleaned_data.get("group")
+        if is_valid:
+            return self.form_valid(form)
+        return self.form_invalid(form)
+
+    def get_form_kwargs(self) -> Dict[str, Any]:
+        """
+        Overrides FormView.get_form_kwargs() to use `request.GET` for data
+        instead `request.POST`. Thes values are used by FormView.get_form()
+        to initialise the form object used by the view.
+        """
+        kwargs = super().get_form_kwargs()
+        data = self.request.GET.copy()
+        for k, v in self.get_form_defaults().items():
+            data.setdefault(k, v)
+        kwargs["data"] = data
+        return kwargs
+
+    def get_form_defaults(self) -> Dict[str, Any]:
+        """
+        Form views have a `get_initial()` method to set initial values on form
+        fields, but because we're ALWAYS providing GET data to the form, those
+        initial values are always overriden. To make up for this,
+        get_form_kwargs() mixes this method's return value into the GET data
+        where no value has been specified, so that the data makes it into the
+        form.
+        """
+        return {}
+
+    def form_valid(self, form: Form) -> HttpResponse:
+        """
+        When the form is valid, fetch results from the API, take any actions
+        based on the result, then render everything to a template.
+        """
+        self.api_result = api_result = self.get_api_result(form)
+        self.process_api_result(form, api_result)
+        context = self.get_context_data(form=form)
+        return self.render_to_response(context)
+
+    def get_meta_title(self) -> str:
+        """
+        Return a string to use the the <title> tag for this view.
+        """
+        title = self.base_title
+        if query := self.form.cleaned_data.get("q", ""):
+            title += ' "' + query.replace('"', "'") + '"'
```

Comment:
Quick Python know-how question here. Is there a reason we wouldn't use a [f-string](https://docs.python.org/3/reference/lexical_analysis.html#f-strings) here?

Reply:
f-string syntax here was triggering a formatting issue in VSCode, because of what is being replaced.

Reply:
I suppose this could be split up into multiple lines, with an f-string used to combine everything on the second line... but, I figured this was simple enough to do on one line.